sycl/CMakeLists.txt

```diff
@@ -39,8 +40,11 @@ target_link_libraries( vecmem_sycl PUBLIC vecmem::core )
 set_target_properties( vecmem_sycl PROPERTIES
    CXX_VISIBILITY_PRESET  "hidden"
    SYCL_VISIBILITY_PRESET "hidden" )
-vecmem_check_sycl_code_compiles( VECMEM_HAVE_SYCL_VISIBILITY_MS_COMPAT
-   "${CMAKE_CURRENT_SOURCE_DIR}/cmake/empty.sycl"
+vecmem_check_sycl_source_compiles( "
+   #include <CL/sycl.hpp>
+   int main() { return 0; }
+   "
+   VECMEM_HAVE_SYCL_VISIBILITY_MS_COMPAT
    CMAKE_FLAGS -DCOMPILE_DEFINITIONS=-fvisibility-ms-compat )
```

Comment:
Not part of the diff, but why does this have to go through `CMAKE_FLAGS`?

Reply:
Because [try_compile(...)](https://cmake.org/cmake/help/latest/command/try_compile.html) is a pain. :frowning: See the description of `CMAKE_FLAGS` under:

https://cmake.org/cmake/help/latest/command/try_compile.html#options

So I just kept the "standard CMake interface" for the SYCL function as well. As broken as it may be...